pkg/agent/agent.go

```diff
@@ -95,5 +128,64 @@ func (a *Agent) Run(healthCheck *metrics.HealthCheck, opts *options.Options) {
 }
 
 func getAgentName(nodeName string) string {
-	return nodeName + "_" + string(uuid.NewUUID())
+	return nodeName + "." + string(uuid.NewUUID())
+}
+
+func (a *Agent) CreateNodeResourceTsp() string {
+	tsp, err := a.craneClient.PredictionV1alpha1().TimeSeriesPredictions("default").Get(context.TODO(), a.GenerateNodeResourceTspName(), metav1.GetOptions{})
+	if err == nil {
+		klog.V(4).Infof("Found old tsp %s in namespace default", a.GenerateNodeResourceTspName())
+		err := a.DeleteNodeResourceTsp()
+		if err != nil {
+			klog.Errorf("Delete old tsp %s with error: %v", a.GenerateNodeResourceTspName(), err)
+			return a.GenerateNodeResourceTspName()
+		}
+	}
+	config, err := a.kubeClient.CoreV1().ConfigMaps("default").Get(context.TODO(), "noderesource-tsp-template", metav1.GetOptions{})
+
+	if err != nil {
+		klog.Exitf("Get noderesource tsp configmap noderesource-tsp-template with error: %v", err)
+	}
+
+	if config == nil {
+		klog.Exitf("Can't get noderesource tsp configmap noderesource-tsp-template")
+	}
+
+	spec := v1alpha12.TimeSeriesPredictionSpec{}
+	err = yaml.Unmarshal([]byte(strings.Replace(config.Data["spec"], "{{nodename}}", a.host, -1)), &spec)
+	if err != nil {
+		klog.Exitf("Convert spec template error: %v", err)
+	}
+
+	n, _ := a.kubeClient.CoreV1().Nodes().Get(context.TODO(), a.host, metav1.GetOptions{})
```

Comment:
Error should be handled

Reply:
Eh... I just think the crane-agent is up, so it should be possible to get nodes normally. Ha, I'll change the code.